### PR TITLE
Remove unwanted deps + modify compat

### PR DESCRIPTION
**Pull request by navidcy**
So it seems that something broke related to BoundaryConditions and the change happened from Oceananigans v0.91.6 to v0.91.7. I added a compat entry here but ideally we'd like to figure out what change happened that causes the test to fail for v0.91.7.

### CODE REVIEW

**Comment by simone-silvestri** [comment]
It was because of https://github.com/CliMA/Oceananigans.jl/pull/3695. In theory we can modify 
https://github.com/CliMA/OrthogonalSphericalShellGrids.jl/blob/8fdd4d9fc03074fe95bf7014775b7addc4958fe0/src/zipper_boundary_condition.jl#L50

to be 
```julia
ZipperBoundaryCondition(sign = 1) = BoundaryCondition(Zipper(), sign)
```
and this should work

**Comment by navidcy** [comment]
Indeed! Done!

**Comment by navidcy** [comment]
Can you give me admin rights on the repo @simone-silvestri?

**Review by simone-silvestri — APPROVED** [review]

File: src/distributed_tripolar_grid.jl [hunk 1]
```diff
@@ -1,52 +1,52 @@
...
+    ranks,
+    inject_halo_communication_boundary_conditions,
+    concatenate_local_sizes
 
 using Oceananigans.Grids: topology
 
 import Oceananigans.DistributedComputations: reconstruct_global_grid
 
-const DistributedTripolarGrid{FT, TX, TY, TZ, A, R, FR, Arch} = OrthogonalSphericalShellGrid{FT, TX, TY, TZ, A, R, FR, <:Tripolar, <:Distributed}
+const DistributedTripolarGrid{FT,TX,TY,TZ,A,R,FR,Arch} = OrthogonalSphericalShellGrid{FT,TX,TY,TZ,A,R,FR,<:Tripolar,<:Distributed}
```
Comment (simone-silvestri) [inline src/distributed_tripolar_grid.jl:13]: why removing the spaces here?

Comment (simone-silvestri) [reply src/distributed_tripolar_grid.jl:13]: and in other places in this file?

Comment (navidcy) [reply src/distributed_tripolar_grid.jl:13]: I think VS code did it! I didn't mean to do that

Comment (navidcy) [reply src/distributed_tripolar_grid.jl:13]: I'll fix it

**Comment by navidcy** [comment]
Seems that https://github.com/CliMA/Oceananigans.jl/pull/3695 introduced a breaking change after all?
Should we bump a minor release in Oceananigans? 

cc @glwagner

**Pull request merged**